### PR TITLE
Add OpenAPI Generator and fix package conflicts

### LINKED ISSUE #27: Feature: Add OpenAPI Generator to Web Frontend
As a developer,
I want to use `npm run gen` command in Web Frontend's path to generate API services to handle backend's endpoints on frontend.

https://openapi-generator.tech/docs/installation#npm

### PR DESCRIPTION
**Pull request by Lukash88**
## Summary

OpenAPI Generator has been added and package conflicts have been fixed.

## Details

**OpenAPI Generator** has been added  to  the application's frontend.
It can  be used by **npm run gen** command in Web Frontend's path to generate API services to handle backend's endpoints on frontend.

## References

No external references needed.

### CODE REVIEW

**Comment by StraykerPL** [comment]
Fill up PR's metadata.

**Comment by StraykerPL** [comment]
Fill up PR's description.

File: binder-web-frontend/package.json [hunk 1]
```diff
@@ -22,13 +23,15 @@
     "@angular/platform-browser-dynamic": "^15.2.3",
     "@angular/router": "^15.2.3",
     "rxjs": "~7.5.0",
+    "semver": "^7.5.4",
```
Comment (StraykerPL) [inline binder-web-frontend/package.json]: What this package is needed for?

File: binder-web-frontend/package.json [hunk 2]
```diff
@@ -22,13 +23,15 @@
...
     "tslib": "^2.3.0",
     "zone.js": "~0.11.4"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "^15.2.4",
+    "@angular-devkit/build-angular": "^15.2.9",
```
Comment (StraykerPL) [inline binder-web-frontend/package.json]: Is it safe to just change single package of Angular, without lifting framework version?

File: binder-web-frontend/package.json [hunk 3]
```diff
@@ -7,8 +7,9 @@
     "build": "ng build",
     "watch": "ng build --watch --configuration development",
     "prepare-husky": "cd .. && husky install binder-web-frontend/.husky",
-    "prettier": "npx prettier --write --ignore-unknown src/"
-  },
+    "prettier": "npx prettier --write --ignore-unknown src/",
+    "gen": "openapi-generator-cli generate -g typescript-angular -i https://localhost:7183/swagger/v1/swagger.json -o ./src/api/code-gen"
```
Comment (StraykerPL) [inline binder-web-frontend/package.json]: Without "code-gun". Just "./src/api".

**Review by StraykerPL — DISMISSED** [review]

File: binder-web-frontend/package.json (hunk 1 above)
Comment (Lukash88) [reply binder-web-frontend/package.json]: After installing **@openapitools/openapi-generator-cli** got issue with vulnerable packages. 
More about you can find [https://github.com/advisories/GHSA-c2qf-rxjj-qqgw](url)
The version in the project was "7.3.8".

Comment (Lukash88) [reply binder-web-frontend/package.json]: I have deleted it and changed versions of **semver** in package-lock.json

File: binder-web-frontend/package.json (hunk 2 above)
Comment (Lukash88) [reply binder-web-frontend/package.json]: It depends but in this case it was unnecessary so I downgrade the package back to previous version.

File: binder-web-frontend/package.json (hunk 3 above)
Comment (Lukash88) [reply binder-web-frontend/package.json]: Resolved 7a76a1c

File: binder-web-frontend/package.json (hunk 1 above)
Comment (Lukash88) [reply binder-web-frontend/package.json]: Resolved  e589e90

File: binder-web-frontend/package.json (hunk 2 above)
Comment (Lukash88) [reply binder-web-frontend/package.json]: Resolved  e589e90

**Comment by StraykerPL** [comment]
Update PR's description after content changes.

**Comment by StraykerPL** [comment]
Lack of resolve info comments in conversations, reference to Wiki for more details.

**Review by Lukash88 — COMMENTED** [review]
All requested changes are addressed

**Review by StraykerPL — APPROVED** [review]

**Pull request merged**